Migrate key import screens to flutter_localizations

buildc0de opened the pull request:
### 🗃 Github Issue

Implement localization using Weblate #1364

### ✅ Checklist

- [x] Github issue details are up to date for people to QA.
- [x] I have tested all my changes.

### 🕵️‍♂️ Notes for Code Reviewer

This PR:
* Migrates key import screens to flutter_localizations
* Adds the concepts of localized errors (enums) instead of `errorMessage` strings
* Adds Hebrew translation for onboarding screens

### 👯‍♀️ Paired with

Nobody.

buildc0de committed:
Configure file generation for flutter_localizations

buildc0de committed:
Define supported locales

buildc0de committed:
Run 'flutter upgrade'

buildc0de committed:
Migrate onboarding screens to flutter_localizations

buildc0de committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (7 of 7 strings)

Translation: SEEDS Light Wallet/Flutter App
Translate-URL: https://translate.hypha.earth/projects/seeds-light-wallet/flutter-app/es/

buildc0de committed:
Translated using Weblate (Portuguese)

Currently translated at 100.0% (7 of 7 strings)

Translation: SEEDS Light Wallet/Flutter App
Translate-URL: https://translate.hypha.earth/projects/seeds-light-wallet/flutter-app/pt/

buildc0de committed:
Translated using Weblate (Portuguese)

Currently translated at 100.0% (7 of 7 strings)

Translation: SEEDS Light Wallet/Flutter App
Translate-URL: https://translate.hypha.earth/projects/seeds-light-wallet/flutter-app/pt/

buildc0de committed:
Translated using Weblate (Portuguese)

Currently translated at 100.0% (7 of 7 strings)

Translation: SEEDS Light Wallet/Flutter App
Translate-URL: https://translate.hypha.earth/projects/seeds-light-wallet/flutter-app/pt/

buildc0de committed:
Organize imports

buildc0de committed:
Organize imports

buildc0de committed:
Add Hebrew as a supported locale

Allows us to begin accepting Hebrew translation contributions in Weblate.

bela-litman committed:
Translated using Weblate (Hebrew)

Currently translated at 100.0% (7 of 7 strings)

Translation: SEEDS Light Wallet/Flutter App
Translate-URL: https://translate.hypha.earth/projects/seeds-light-wallet/flutter-app/he/

buildc0de committed:
Migrate key import screens to flutter_localizations

buildc0de committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (19 of 19 strings)

Translation: SEEDS Light Wallet/Flutter App
Translate-URL: https://translate.hypha.earth/projects/seeds-light-wallet/flutter-app/es/

buildc0de committed:
Translated using Weblate (Portuguese)

Currently translated at 100.0% (19 of 19 strings)

Translation: SEEDS Light Wallet/Flutter App
Translate-URL: https://translate.hypha.earth/projects/seeds-light-wallet/flutter-app/pt/

buildc0de committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (19 of 19 strings)

Translation: SEEDS Light Wallet/Flutter App
Translate-URL: https://translate.hypha.earth/projects/seeds-light-wallet/flutter-app/es/

buildc0de committed:
Merge master into weblate

buildc0de committed:
Update Runner after adding in_app_review

RaulUrtecho commented on lib/screens/authentication/import_key/import_key_errors.dart:
```diff
@@ -0,0 +1,20 @@
+import 'package:flutter/widgets.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+
+enum ImportKeyError { InvalidPrivateKey, NoAccountsFound, UnableToLoadAccount, NoPublicKeyFound }
+
+extension LocalizedImportKeyError on ImportKeyError {
+  String localizedDescription(BuildContext context) {
+    final localization = AppLocalizations.of(context)!;
+    switch (this) {
+      case ImportKeyError.InvalidPrivateKey:
+        return localization.importKeyInvalidPrivateKeyError;
+      case ImportKeyError.NoAccountsFound:
+        return localization.importKeyNoAccountsFoundError;
+      case ImportKeyError.UnableToLoadAccount:
+        return localization.importKeyUnableToLoadAccountError;
+      case ImportKeyError.NoPublicKeyFound:
+        return localization.importKeyNoPublicKeyFoundError;
+    }
+  }
+}
```
Cool

RaulUrtecho commented on lib/l10n/app_en.arb:
```diff
@@ -5,5 +5,17 @@
     "onboardingCampaignsSubtitle": "Participate and vote directly on social and\nenvironmental impact projects you care\nabout.",
     "onboardingEconomyTitle": "Regenerative\nEconomy",
     "onboardingEconomySubtitle": "Unite with a global movement of\norganizations and people to regenerate our\nplanet and heal our economy.",
-    "onboardingJoinButtonTitle": "Join Now"
```
So basically we will have one file per language. 
Since these monolithic files will grow indefinitely, I consider it a good practice to put a comment that indicates the following lines to which screen they belong in order to make it easier to search and read these files.

buildc0de replied:
Yes! Comments would have been nice 🤓 The issue is that .arb files are JSON files and they can't contain comments. Here is the compilation error:
> Generating synthetic localizations package failed with 1 error:
>
> Exception: The arb file .. /seeds_light_wallet/lib/l10n/app_en.arb has the following formatting issue: 
> FormatException: Unexpected character (at line 10, character 5)
>     /// Import Key Screen
>     ^

RaulUrtecho reviewed (commented):
For new translations of each screen added, please add a comment line first. Example:
```
"onboardingJoinButtonTitle": "Join Now",
"globalUnknownError": "Oops, something went wrong",
/// Import Key Screen
"importKeySearchButtonTitle": "Search",
"importKeyPrivateKeyFieldPlaceholder": "Private Key",
...
```

bela-litman committed:
Translated using Weblate (Hebrew)

Currently translated at 100.0% (19 of 19 strings)

Translation: SEEDS Light Wallet/Flutter App
Translate-URL: https://translate.hypha.earth/projects/seeds-light-wallet/flutter-app/he/

buildc0de committed:
Translated using Weblate (Hebrew)

Currently translated at 100.0% (19 of 19 strings)

Translation: SEEDS Light Wallet/Flutter App
Translate-URL: https://translate.hypha.earth/projects/seeds-light-wallet/flutter-app/he/

gguijarro-c-chwy commented on lib/screens/authentication/import_key/import_key_errors.dart:
```diff
@@ -0,0 +1,20 @@
+
+enum ImportKeyError { InvalidPrivateKey, NoAccountsFound, UnableToLoadAccount, NoPublicKeyFound }
+
+extension LocalizedImportKeyError on ImportKeyError {
```
👍
